Vendor code not in remote repository

Hi,
i try to migrate my project from *gb* to *glide*. On localhost, everything works as it should. But if i clone my project, vendor directory doesn't contains any code : 

```bash
$ git clone https://github.com/nlamirault/abraracourcix
git clone https://github.com/nlamirault/abraracourcix
Cloning into 'abraracourcix'...
remote: Counting objects: 1979, done.
remote: Compressing objects: 100% (85/85), done.
remote: Total 1979 (delta 35), reused 0 (delta 0), pack-reused 1884
Receiving objects: 100% (1979/1979), 6.91 MiB | 2.91 MiB/s, done.
Resolving deltas: 100% (488/488), done.
Checking connectivity... done.
$ find abraracourcix/vendor/ -name "*.go"
```

According to commit message [1], i can see glide use SubProjects system.
But if i try to update submodules i've got an error : 

```bash
$ glide install
[INFO] Lock file (glide.lock) does not exist. Performing update.
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/boltdb/bolt.
[INFO] Fetching updates for github.com/garyburd/redigo.
[INFO] Fetching updates for github.com/hashicorp/logutils.
[INFO] Fetching updates for github.com/syndtr/goleveldb.
[INFO] Fetching updates for github.com/labstack/echo.
[INFO] Fetching updates for gopkg.in/mgo.v2.
[INFO] Resolving imports
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/api (1)
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/io (1)
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/logging (1)
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/storage (1)
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/version (1)
[INFO] Not found in vendor/: github.com/nlamirault/abraracourcix/api/v1 (1)
[ERROR] Error scanning github.com/labstack/echo/middleware: open /tmp/TMP/abraracourcix/vendor/github.com/labstack/echo/middleware: no such file or directory
[ERROR] Error scanning github.com/garyburd/redigo/redis: open /tmp/TMP/abraracourcix/vendor/github.com/garyburd/redigo/redis: no such file or directory
[ERROR] Error scanning github.com/syndtr/goleveldb/leveldb: open /tmp/TMP/abraracourcix/vendor/github.com/syndtr/goleveldb/leveldb: no such file or directory
[ERROR] Error scanning gopkg.in/mgo.v2/bson: open /tmp/TMP/abraracourcix/vendor/gopkg.in/mgo.v2/bson: no such file or directory
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for github.com/nlamirault/abraracourcix.
[INFO] Setting references for remaining imports
[INFO] Project relies on 7 dependencies.
An Error has occured
```

How can i solve this problem ?
is it possible not to use Glide subprojects system but rather the source code dependencies?

Thanks

[1] : https://github.com/nlamirault/abraracourcix/commit/34e291a1b4668398910d1171e8df43a6662da7d5

I update the test : 

```
$ echo $GOPATH
/home/nlamirault/Apps/golang
$ cd $GOPATH/src/github.com/nlamirault
$ git clone https://github.com/nlamirault/abraracourcix
Cloning into 'abraracourcix'...
remote: Counting objects: 1985, done.
remote: Compressing objects: 100% (91/91), done.
remote: Total 1985 (delta 39), reused 0 (delta 0), pack-reused 1884
Receiving objects: 100% (1985/1985), 6.91 MiB | 1.04 MiB/s, done.
Resolving deltas: 100% (492/492), done.
Checking connectivity... done.
$ cd abraracourcix
$ glide install
glide install
[INFO] Lock file (glide.lock) does not exist. Performing update.
[INFO] Downloading dependencies. Please wait...
[INFO] Fetching updates for gopkg.in/mgo.v2.
[INFO] Fetching updates for github.com/boltdb/bolt.
[INFO] Fetching updates for github.com/labstack/echo.
[INFO] Fetching updates for github.com/syndtr/goleveldb.
[INFO] Fetching updates for github.com/garyburd/redigo.
[INFO] Fetching updates for github.com/hashicorp/logutils.
[INFO] Resolving imports
[ERROR] Error scanning github.com/labstack/echo/middleware: open /home/nlamirault/Apps/golang/src/github.com/nlamirault/abraracourcix/vendor/github.com/labstack/echo/middleware: no such file or directory
[ERROR] Error scanning github.com/garyburd/redigo/redis: open /home/nlamirault/Apps/golang/src/github.com/nlamirault/abraracourcix/vendor/github.com/garyburd/redigo/redis: no such file or directory
[ERROR] Error scanning github.com/syndtr/goleveldb/leveldb: open /home/nlamirault/Apps/golang/src/github.com/nlamirault/abraracourcix/vendor/github.com/syndtr/goleveldb/leveldb: no such file or directory
[ERROR] Error scanning gopkg.in/mgo.v2/bson: open /home/nlamirault/Apps/golang/src/github.com/nlamirault/abraracourcix/vendor/gopkg.in/mgo.v2/bson: no such file or directory
[INFO] Downloading dependencies. Please wait...
[INFO] Setting references for remaining imports
[INFO] Project relies on 6 dependencies.
An Error has occured

```

--- a/dependency/resolver.go
+++ b/dependency/resolver.go
@@ -400,7 +400,7 @@ func (r *Resolver) resolveImports(queue *list.List) ([]string, error) {
 			msg.Debug("Using Iterative Scanning for %s", dep)
 			imps, err = IterativeScan(vdep)
 			if err != nil {
-				msg.Err("Error scanning %s: %s", dep, err)
+				msg.Err("Iterative scanning error %s: %s", dep, err)
 				continue
 			}
 		} else if err != nil {

--- a/repo/installer.go
+++ b/repo/installer.go
@@ -329,8 +329,23 @@ func (m *MissingPackageHandler) NotFound(pkg string) (bool, error) {
 	// This package may have been placed on the list to look for when it wasn't
 	// downloaded but it has since been downloaded before coming to this entry.
 	if _, err := os.Stat(dest); err == nil {
-		msg.Debug("Found %s", dest)
-		return true, nil
+		// Make sure the location contains files. It may be an empty directory.
+		empty, err := gpath.IsDirectoryEmpty(dest)
+		if err != nil {
+			return false, err
+		}
+		if empty {
+			msg.Warn("%s is an existing location with no files. Fetching a new copy of the dependency.", dest)
+			msg.Debug("Removing empty directory %s", dest)
+			err := os.RemoveAll(dest)
+			if err != nil {
+				msg.Debug("Installer error removing directory %s: %s", dest, err)
+				return false, err
+			}
+		} else {
+			msg.Debug("Found %s", dest)
+			return true, nil
+		}
 	}
 
 	msg.Info("Fetching %s into %s", pkg, m.destination)

--- a/repo/vcs.go
+++ b/repo/vcs.go
@@ -55,6 +55,17 @@ func VcsUpdate(dep *cfg.Dependency, dest, home string, cache, cacheGopath, useGo
 		_, err = v.DetectVcsFromFS(dest)
 		if updateVendored == false && empty == false && err == v.ErrCannotDetectVCS {
 			msg.Warn("%s appears to be a vendored package. Unable to update. Consider the '--update-vendored' flag.\n", dep.Name)
+		} else if updateVendored == false && empty == true && err == v.ErrCannotDetectVCS {
+			msg.Warn("%s is an empty directory. Fetching a new copy of the dependency.", dep.Name)
+			msg.Debug("Removing empty directory %s", dest)
+			err := os.RemoveAll(dest)
+			if err != nil {
+				return err
+			}
+			if err = VcsGet(dep, dest, home, cache, cacheGopath, useGopath); err != nil {
+				msg.Warn("Unable to checkout %s\n", dep.Name)
+				return err
+			}
 		} else {
 
 			if updateVendored == true && empty == false && err == v.ErrCannotDetectVCS {

